v0/util/cache.js

```diff
@@ -17,12 +17,20 @@ class Cache {
 
     return storeFunction().then(result => {
       // store in cache if the value is valid, else skip
+      if (result && typeof result === "object") {
```

Comment:
This approach fails if we were to store objects instead of primitive values in the cache, it'd fail.
Update the signature of the store function to take two values, namely, `value` and `age` (optional).

Reply:
Okay, but, why would it fail with the current implementation? As we can store objects also as a value here.

Reply:
It'd fail if the `result` is an objection and doesn't follow the structure as expected in Ln 20-22.